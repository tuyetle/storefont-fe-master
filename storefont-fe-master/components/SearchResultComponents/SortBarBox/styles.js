import variables from '~/styles/variables';
import { css } from 'glamor';

export default {
  contain: css({
    display: 'inline-block',
    color: variables.$colorFacetSearchNavItem,
    fontSize: '0.750em',
    width: '1.6875em',
    height: '1.25em',
    border: 'none',
    paddingTop: '0.1em',
    verticalAlign: 'middle',
    marginRight: '1.25em',
    marginTop: '1em',

    '& a': {
      paddingLeft: '1em',
      '@media (max-width: 767px)': {
        display: 'inline-block',
        width: '100%',
        height: '1.75em',
        backgroundImage: `url('./static/assets/img/utilityIcons/ic-sort.svg')`,
        backgroundRepeat: 'no-repeat',
        paddingLeft: '1em',
        textIndent: '-100em',
        marginTop: '-0.2em',
      },
    },

    '@media (min-width: 768px)': {
      width: '14.7em',
      height: '1.938em',
      border: `solid 0.5px ${variables.$colorFacetSearchNavItem}`,
      background: 'none',
      verticalAlign: 'none',
      marginRight: '0',
      marginTop: '0',
    },

  }),
  sortBox: css({
    position: 'relative',
    top: '-0.5em',
    paddingTop: '0.3em',
    paddingBottom: '0.3em',
    width: '14.7em',
    fontSize: '0.750em',
    color: variables.$colorFacetSearchNavItem,
    backgroundColor: 'white',
    boxShadow: variables.$boxShadowListView,
    '& button': {
      paddingLeft: '0.813em',
      '&:active': {
        color: variables.$colorFacetSearchNavItem,
      },
    },
  }),
};
