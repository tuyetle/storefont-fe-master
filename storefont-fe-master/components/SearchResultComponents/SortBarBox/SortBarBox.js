import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DropdownItem } from 'reactstrap';
import { translate } from 'react-i18next';
import DropdownButton from '~/components/common/DropdownButton/DropdownButton';
import { SORT_RUSULT_OPTIONS } from '~/config/config';
import styles from './styles';

export class SortBarBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: SORT_RUSULT_OPTIONS[0].name,
    };
  }
  selectHandler = (e) => {
    this.setState({
      value: e.target.value,
    });
    this.props.onSortRusultChanged(e.target.value);
  };
  render() {
    return (
      <DropdownButton toggleTag="a" className={`${styles.contain}`} label={this.props.t(this.state.value)}>
        <section className={styles.sortBox}>
          {
            SORT_RUSULT_OPTIONS.map((opt, i) => (
              <DropdownItem key={+i} onClick={this.selectHandler} value={opt.name}>
                {this.props.t(opt.name)}
              </DropdownItem>
            ))
          }
        </section>
      </DropdownButton>
    );
  }
}

SortBarBox.propTypes = {
  t: PropTypes.func.isRequired,
  onSortRusultChanged: PropTypes.func,
};

SortBarBox.defaultProps = {
  onSortRusultChanged: null,
};
export default translate(['searchResult'])(SortBarBox);
