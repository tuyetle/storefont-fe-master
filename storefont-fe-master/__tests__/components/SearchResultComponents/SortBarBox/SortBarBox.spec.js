import React from 'react';
import renderer from 'react-test-renderer';
import { SortBarBox } from '~/components/SearchResultComponents/SortBarBox/SortBarBox';

describe('(Component) SortBarBox', () => {
  let component;
  let props;
  let spyCallback;
  beforeEach(() => {
    props = {
      t(value) { return value; },
      onSortRusultChanged() {},
    };
    spyCallback = jest.spyOn(props, 'onSortRusultChanged');
    component = mount(<SortBarBox {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call callback when select', () => {
    const dropdown = component.find('button');
    dropdown.first().simulate('click');
    expect(spyCallback).toHaveBeenCalled();
  });
});
