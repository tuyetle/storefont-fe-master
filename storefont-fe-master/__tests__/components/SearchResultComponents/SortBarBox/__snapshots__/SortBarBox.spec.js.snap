// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SortBarBox should render conrectly 1`] = `
<div
  className="css-zwxkl8 css-sl3pwm"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="css-12368be css-1rce7jy"
      data-toggle="dropdown"
      onClick={[Function]}
    >
      oldToNew
    </a>
    <div
      aria-hidden={true}
      className="null pull-right dropdown-menu"
      role="menu"
      tabIndex="-1"
    >
      <section
        className={
          Object {
            "data-css-4n0gx5": "",
          }
        }
      >
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
          value="oldToNew"
        >
          oldToNew
        </button>
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
          value="newToOld"
        >
          newToOld
        </button>
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
          value="lowToHighPrice"
        >
          lowToHighPrice
        </button>
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
          value="highToLowPrice"
        >
          highToLowPrice
        </button>
      </section>
    </div>
  </div>
</div>
`;
